numpy/_core/getlimits.py

```diff
@@ -516,6 +516,11 @@ def __new__(cls, dtype):
         obj = cls._finfo_cache.get(dtype)
         if obj is not None:
             return obj
+        if hasattr(dtype, "finfo") and issubclass(dtype.type, numeric.inexact):
+            # User-defined dtype with finfo support
+            finfo_obj = dtype.finfo()
+            obj = object.__new__(cls)._init_from_finfo_obj(dtype, finfo_obj)
```

Comment:
With the class method suggested below, this becomes `return cls._from_dtype_finfo(dtype)`